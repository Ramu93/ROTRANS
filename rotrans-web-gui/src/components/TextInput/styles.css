.text-div {
  margin-right: 30px;
}

.text-input {
  width: 432px;
  height: 36px;
  border: 1.5px solid #ededed;
  border-radius: 4px;
  padding-left: 8px;
}

.text-input-full {
  width: 100%;
}

.text-input:focus {
  outline-style: none;
}

.text-label {
  font-size: 15px;
  line-height: 32px;
  color: #333333;
}

.text-button {
  margin-left: 8px;
  height: 40px;
  width: 40px;
  background-color: #f3f3f3;
  border-radius: 4px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  border: 0;
  cursor: pointer;
}

.text-combo {
  display: flex;
  flex-direction: row;
  align-items: center;
}

.text-hide {
  filter: blur(0.2rem) !important;
}