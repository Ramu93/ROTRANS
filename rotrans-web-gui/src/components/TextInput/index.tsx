import React from "react";
import "./styles.css";

interface Props {
  ref?: any;
  label: string;
  type?: string;
  placeholder?: string;
  value?: string;
  br?: boolean;
  hide?: boolean;
  button?: any;
  onButtonClick?: (
    event: React.MouseEvent<HTMLButtonElement, MouseEvent>
  ) => void | undefined;
  onChange?: ((event: React.ChangeEvent<HTMLInputElement>) => void) | undefined;
  onBlur?: ((event: React.FocusEvent<HTMLInputElement>) => void) | undefined;
  full?: boolean;
  onFocus?: ((event: React.FocusEvent<HTMLInputElement>) => void) | undefined;
  dropdown?: any;
  required?: boolean;
  step?: number | string | undefined;
  name?: string;
}

const TextInput: React.FC<Props> = ({
  ref,
  label,
  type,
  placeholder,
  value,
  onChange,
  onBlur,
  button,
  onButtonClick,
  hide,
  br,
  full,
  onFocus,
  dropdown,
  required,
  step,
  name,
}) => {
  return (
    <>
      <div data-testid="textInputMainDiv" className={button ? "text-div" : ""}>
        <span data-testid="textInputLabel" className="text-label">
          {label}
        </span>
        {required && <span style={{ color: "red" }}>*</span>}
        <div className="text-combo">
          <input
            ref={ref}
            className={
              hide
                ? "text-input text-hide"
                : full
                ? "text-input text-input-full"
                : "text-input"
            }
            type={type ? type : "text"}
            placeholder={placeholder}
            onChange={onChange}
            value={value}
            onBlur={onBlur}
            onFocus={onFocus}
            data-testid="textInput"
            step={type === "number" ? step : undefined}
            name={name}
          />
          {button && (
            <button
              data-testid="textInputButton"
              className="text-button icon"
              onClick={onButtonClick}
            >
              {button}
            </button>
          )}
        </div>
      </div>
      {dropdown}
      {br && <br data-testid="textInputBr" />}
    </>
  );
};

export default TextInput;
