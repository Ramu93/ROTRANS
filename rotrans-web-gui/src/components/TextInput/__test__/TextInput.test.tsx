import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import TextInput from "..";

describe("TextInput component", () => {
  afterEach(cleanup);

  it("Rendered without button and hidden prop", () => {
    const { getByTestId } = render(<TextInput label="ABC" value="123" />);
    expect(getByTestId("textInputLabel")).toHaveTextContent("ABC");
    expect(getByTestId("textInput")).toHaveValue("123");
    expect(getByTestId("textInputMainDiv")).not.toHaveClass("text-div");
    expect(getByTestId("textInput")).not.toHaveClass("text-hide");
  });

  it("Rendered with button", () => {
    const { getByTestId } = render(
      <TextInput
        label="ABC"
        value="123"
        button={<img src={require("../../../assets/svg/lock.svg")} />}
      />
    );
    expect(getByTestId("textInputButton")).toBeVisible();
    expect(getByTestId("textInputMainDiv")).toHaveClass("text-div");
  });

  it("Rendered with breakline", () => {
    const { getByTestId } = render(<TextInput label="ABC" value="123" br />);
    expect(getByTestId("textInputBr")).toBeVisible();
  });

  it("Rendered with hidden prop", () => {
    const { getByTestId } = render(<TextInput label="ABC" value="123" hide />);
    expect(getByTestId("textInput")).toHaveClass("text-hide");
  });

  it("Matches snapshot 1 - without button", () => {
    const tree = renderer
      .create(<TextInput label="ABC" value="123" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Matches snapshot 2 - with button", () => {
    const tree = renderer
      .create(
        <TextInput
          label="ABC"
          value="123"
          button={<img src={require("../../../assets/svg/lock.svg")} />}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Matches snapshot 3 - with button, hidden prop, br", () => {
    const tree = renderer
      .create(
        <TextInput
          label="ABC"
          value="123"
          button={<img src={require("../../../assets/svg/lock.svg")} />}
          hide
          br
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
