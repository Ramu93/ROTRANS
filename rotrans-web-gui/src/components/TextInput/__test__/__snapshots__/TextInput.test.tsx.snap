// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput component Matches snapshot 1 - without button 1`] = `
<div
  className=""
  data-testid="textInputMainDiv"
>
  <span
    className="text-label"
    data-testid="textInputLabel"
  >
    ABC
  </span>
  <div
    className="text-combo"
  >
    <input
      className="text-input"
      data-testid="textInput"
      type="text"
      value="123"
    />
  </div>
</div>
`;

exports[`TextInput component Matches snapshot 2 - with button 1`] = `
<div
  className="text-div"
  data-testid="textInputMainDiv"
>
  <span
    className="text-label"
    data-testid="textInputLabel"
  >
    ABC
  </span>
  <div
    className="text-combo"
  >
    <input
      className="text-input"
      data-testid="textInput"
      type="text"
      value="123"
    />
    <button
      className="text-button icon"
      data-testid="textInputButton"
    >
      <img
        src={
          Object {
            "ReactComponent": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "__esModule": true,
            "default": "lock.svg",
          }
        }
      />
    </button>
  </div>
</div>
`;

exports[`TextInput component Matches snapshot 3 - with button, hidden prop, br 1`] = `
Array [
  <div
    className="text-div"
    data-testid="textInputMainDiv"
  >
    <span
      className="text-label"
      data-testid="textInputLabel"
    >
      ABC
    </span>
    <div
      className="text-combo"
    >
      <input
        className="text-input text-hide"
        data-testid="textInput"
        type="text"
        value="123"
      />
      <button
        className="text-button icon"
        data-testid="textInputButton"
      >
        <img
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "lock.svg",
            }
          }
        />
      </button>
    </div>
  </div>,
  <br
    data-testid="textInputBr"
  />,
]
`;
